### PR TITLE
prime-factors: use cases of x-common

### PR DESCRIPTION
**Pull request by petertseng**
https://github.com/exercism/x-common/pull/513

---

reviewers: your opinion will be appreciated on whether to incorporate the test descriptions into the file.

1. Do nothing.
2. Include as comments.
3. Include as an element of the tuple, then use `it` on them so thye become test descriptions.

### CODE REVIEW

**Comment by petertseng** [comment]
(My inclination is go for 3, but I didn't want to spend the time if nobody else cared)

**Comment by abo64** [comment]
3 is more consistent with most other test suites, but it is indeed questionable whether it is worth the effort.

Maybe create a separate issue, but then for checking all test suites if they include the descriptions?

**Comment by petertseng** [comment]
I don't necessarily think we need descriptions for all exercises - roman-numerals doesn't really need any https://github.com/exercism/xhaskell/blob/master/exercises/roman-numerals/test/Tests.hs - unless https://github.com/exercism/x-common/pull/451 comes up with something good.

**Comment by rbasso** [comment]
Considering that one of the goals of Exercism seems to be keeping some degree of cross-language homogeneity, I would favor option 3, even in cases where the descriptions do not add too much to the tests.

File: exercises/prime-factors/test/Tests.hs [hunk 1]
```diff
@@ -14,22 +14,16 @@ specs = describe "prime-factors" $
...
-            , (          9,             [3, 3] )
-            , (         27,          [3, 3, 3] )
-            , (        625,       [5, 5, 5, 5] )
-            , (     901255,   [5, 17, 23, 461] )
-            , (93819012551, [11, 9539, 894119] ) ]
+    cases = [ ("no factors",                              1,                 [] )
+            , ("prime number",                            2,                [2] )
+            , ("square of a prime",                       9,             [3, 3] )
+            , ("cube of a prime",                         8,          [2, 2, 2] )
+            , ("product of primes and non-primes",       12,          [2, 2, 3] )
```
Comment (petertseng) [inline exercises/prime-factors/test/Tests.hs]: someone who is looking closely will see that there is a shortcut being taken in the indentation: the comma overhangs the 3 two lines below.

**Pull request merged**